src/LazyOperations/InverseLinearMap.jl

```diff
@@ -0,0 +1,333 @@
+import Base: *, ∈, isempty
+
+export InverseLinearMap,
+       an_element,
+       constraints_list
+
+"""
+   InverseLinearMap{N<:Real, S<:LazySet{N}, NM, MAT<:AbstractMatrix{NM}} <: AbstractAffineMap{N, S}
+
+Given a linear transformation ``M``, this type represents the linear
+transformation ``M⁻¹⋅S`` of a convex set ``S``.
+
+### Fields
+
+- `M` -- matrix/linear map
+- `X` -- convex set
+
+### Notes
+
+This type is parametric in the elements of the inverse linear map, `NM`, which is
+independent of the numeric type of the wrapped set (`N`).
+Typically `NM = N`, but there may be exceptions, e.g., if `NM` is an interval
+that holds numbers of type `N`, where `N` is a floating point number type such
+as `Float64`.
+
+### Examples
+
+For the examples we create a ``3×3`` matrix and a unit three-dimensional square.
+
+```jldoctest constructors
+julia> A = [1 2 3; 2 3 1; 3 1 2]; X = BallInf([0, 0, 0], 1); Y = BallInf([0], 1);
+```
+
+```jldoctest constructors
+julia> ilm = InverseLinearMap(A, X)
+InverseLinearMap{Int64,BallInf{Int64,Array{Int64,1}},Int64,Array{Int64,2}}([1 2 3; 2 3 1; 3 1 2], BallInf{Int64,Array{Int64,1}}([0, 0, 0], 1))
+```
+
+Applying a linear map to a `InverseLinearMap` object combines the two maps into
+a single `InverseLinearMap` instance.
+Again we can make use of the conversion for convenience.
+
+```jldoctest constructors
+julia> B = transpose(A); ilm2 = InverseLinearMap(B, ilm)
+InverseLinearMap{Int64,BallInf{Int64,Array{Int64,1}},Int64,Array{Int64,2}}([14 11 11; 11 14 11; 11 11 14], BallInf{Int64,Array{Int64,1}}([0, 0, 0], 1))
+
+julia> ilm2.M == B*A
+true
+```
+
+The application of a `InverseLinearMap` to a `ZeroSet` or an `EmptySet` is
+simplified automatically.
+
+```jldoctest constructors
+julia> InverseLinearMap(A, ZeroSet{Int}(3))
+ZeroSet{Int64}(3)
+```
+"""
+struct InverseLinearMap{N<:Real, S<:LazySet{N},
```

Comment:
can you try removing `N<:Real` here ? i don't see a compelling reason to do so since it is a new type.

Reply:
Yes. I'm fine with the change.